### DISCUSSION TITLE
bS Cookie Settings Plugin sufficient?

### DISCUSSION THREAD

**Discussion by kreincke** [bootscore/bootscore#387] [Q&A]
Dear friends,

If I understood the code and demo of the **bs cookie plugin** correctly, then the user has either the option to accept all cookies or to reject all cookies. Did I understand that correctly?

As far as I can see, that's insufficient for Europe (or at least in Germany). We must offer our readers the option to refuse the setting of single cookies. How can we deal with this requirement in a bootScore based site?

Additionally, how does bootScore enforce WordPress and the other plugins NOT to write cookies if the user rejected the question? Does the site developer have to support this 'You-Must-Not-Write-Process'? 

A similar question: Does bootScore successfully enforce Wordpress and the plugins  NOT to write cookies before the user has agreed to do so? 

Finally, who decides what's shown if the user rejected the cookie writings? Is that an immanent feature of WordPress, of the Plugings, and of bootScore? - Or must the site developer himself prepare a fitting reaction?

In the hope not having annoyed you by such details
KR

**Comment by kreincke** [comment]
After having spent a night with this issue, I think I know - at least for Germany -, that the solution of the bootscore plugin is correct 
* with respect to the fact that it allows to actively respect / reject groups of cookies  The point of the German legal decisions is not, that each single cookie must become rejectable. The point is, that the user must actively approve the usage (by checking a box etc. but NOT by preselected boxes).
* with respect to allow the theme developer to group the cookies in technical necessary and technical not necessary cookies.

This is the best source I found: [https://www.e-recht24.de/artikel/datenschutz/8451-hinweispflicht-fuer-cookies.html](https://www.e-recht24.de/artikel/datenschutz/8451-hinweispflicht-fuer-cookies.html).

During the next week, I will test whether rejecting the permission to write unnecessary cookies by using the bs cookie plugin indeed prevents the writing plugins from writing the cookies.

Wherever I look, I see the excellent work the bootscore developer team does. Again and again: thanks KR

**Comment by crftwrk** [comment]
GDPR is an EU law and everywhere the same. No matter if you are from Spain, Netherlands, Greece, Sweden, Poland, Germany or somewhere else in the EU.

WordPress and some plugins uses cookies for their functionality. These are mandatory cookies and it's not the job of bS Cookie Settings to manage them. You can block them in your browser settings. But of course, you cannot log in or using a language switcher for example anymore, because language preferences are stored in a cookie.

You can manage additional Cookies like a Google Analytics tracker as described here https://bootscore.me/documentation/plugin/bs-cookie-settings/#Block_manage_scripts and ask users if they allow to track or not. This is what all GDPR cookie scripts do.
